### ISSUE TITLE
Debugger Starts but Doesn't Get Response

### ISSUE THREAD

**Issue by JordanEverett** [intersystems-community/vscode-objectscript#1038]
Type: <b>Bug</b>

I am on HealthShare 2017.2 build 865_3_20793U and am having issues getting my VSCode to debug through my CSP Web Gateway. If I go directly to my server, then it is fine, but if I go through my gateway it will start, but never seem to get a response.

I have verified that Websockets are configured on my CSP Gateway server. I also have Websockets configured on my test server.

Is there something that I'm missing in my CSP Gateway configuration that would prevent this from working?

Extension version: 2.2.0
VS Code version: Code 1.74.2 (e8a3071ea4344d9d48ef8a4df2c097372b0c5161, 2022-12-20T10:29:14.590Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No

**Comment by isc-bsaviano** [comment]
@JordanEverett Are you setting breakpoints? If not the debugger is just going to run the method/routine until completion without stopping. Also, make sure you're debugging on the right server. VS Code will start a debug session with the server of the file in the active editor (the tab that the user is focused on).

**Comment by JordanEverett** [comment]
Yes I do have a breakpoint set. I also only have one connection setting set in my workspace. 

If I was to start the debug, then it will act like nothing happened, but if I clicked it again, then two would pop up.

Example:

![image](https://user-images.githubusercontent.com/61211205/210605949-3f4431bd-31d7-4e8b-855f-970752472ab0.png)

It'll just keep spinning until I exit out.

**Comment by isc-bsaviano** [comment]
@JordanEverett Can you exit those sessions, follow the steps outlined [here](https://intersystems-community.github.io/vscode-objectscript/rundebug/#troubleshooting-debugger-issues) and send me the output?

**Comment by isc-bsaviano** [comment]
@JordanEverett Maybe this has something to do with HTTPS. I tried to reproduce your problem using HTTPS but I didn't get as far as you. I didn't have anything in my debug global. Are you using a self-signed certificate? That doesn't seem like it would be a problem since you were able to establish the websocket connection but it can't hurt to ask.

**Comment by JordanEverett** [comment]
I tried going through HTTP but still through my gateway server and I had no luck. The only thing I can really think of now is that I'm missing some sort of configuration setting on my Gateway server, but I haven't found any documentation that could be leading me to a cause...any ideas?

**Comment by isc-bsaviano** [comment]
@JordanEverett I'm glad it's not an HTTPS problem. I don't know what setting you could be missing, Since you can reproduce the problem using HTTP you can use Wireshark to view the packets being sent back and forth. You can compare that to the log global and see at what point debugging is failing. If it's at the same spot every time then I can look in the source code and see what the extension is trying to do at that point.

**Comment by JordanEverett** [comment]
I figured it out! It turned out to be a SSL/TLS misconfiguration from my gateway to my server. The client to the gateway was fine, but the encryption between the server and gateway was the issue. Thank you for your assistance on this!
